Cargo.toml

```diff
@@ -33,10 +34,18 @@ features = [
 atty = "0.2.14"
 
 [target.'cfg(any(unix, target_os = "wasi"))'.dev-dependencies]
+rustix = { version = "0.38.0", features = ["termios", "stdio"] }
 libc = "0.2.110"
 
-[target.'cfg(not(any(windows, target_os = "hermit", target_os = "unknown")))'.dev-dependencies]
-rustix = { version = "0.38.0", features = ["stdio"] }
-
 [target.'cfg(windows)'.dev-dependencies]
 tempfile  = "3"
+
+[features]
+default = ["std", "libc"]
+std = []
```

Comment:
What does this feature do? Just enabling it by itself does not pull anything in.

If the features should represent three states this crate can be in:
* No rustix (use libc)
* Rustix but no_std
* Rustix with std.

... Then I would consider defining the features like this:
```
[features]
default = []

# Implicit `rustix` feature by the optional dependency.
# Switches from `libc` to `rustix` backend.

# Enabling `std` makes is-terminal use the `rustix` backend, with std support.
std = ["rustix", "rustix/std"]
```

The combination `libc` + `std` does not seem to do anything differently compared to just `libc` from what I can see in the current PR code.

But I have to say I'm shooting a bit from the hip here. I don't know why this crate needs two backends. Out of curiosity, why is `rustix` wanted? Does it yield more accurate results compared to libc in some situations?